docs/built-in-models/encrypted_dataframe.md

```diff
@@ -0,0 +1,111 @@
+- **String Enum**: String columns are mapped to integers starting from 1. This mapping is stored and later used for de-quantization. If the number of unique strings exceeds 15, a `ValueError` is raised.
+
+## Supported Operations on Encrypted Data-frames
+
+Encrypted DataFrames support a subset of operations that are available for pandas DataFrames. The following operations are currently supported:
+
+- `merge`: left or right join two data-frames
+
+<!--pytest-codeblocks:cont-->
+
+```python
+df_right = """index,day,time,size
+2,Thur,Lunch,2
+5,Sat,Dinner,3
+9,Sun,Dinner,2"""
+
+# Encrypt the DataFrame
+df_encrypted2 = client.encrypt_from_pandas(pandas.read_csv(StringIO(df_right)))
+
+df_encrypted_merged = df_encrypted.merge(df_encrypted2, how="left", on="index")
+```
+
+> **Important**: The merge operation on Encrypted DataFrames can be **securely** performed on a third-party server. This means that the server can execute the merge without ever having access to the unencrypted data. The server only requires the encrypted DataFrames.
+
+## Serialization of Encrypted Data-frames
+
+Encrypted `DataFrame` objects can be serialized to a file format for storage or transfer. When serialized, they contain the encrypted data and [evaluation keys](../getting-started/concepts.md#cryptography-concepts) necessary to perform computations.
+
+### Saving and loading Data-frames
+
+To save or load an encrypted `DataFrame` from a file, use the following commands:
+
+<!--pytest-codeblocks:cont-->
+
+```python
+# Save
+df_encrypted_merged.save("df_encrypted_merged")
+
+# Load
+df_encrypted_merged = load_encrypted_dataframe("df_encrypted_merged")
+
+# Decrypt the DataFrame
+df_decrypted = client.decrypt_to_pandas(df_encrypted)
+```
+
+## Error Handling
+
+The library is designed to raise specific errors when encountering issues during the pre-processing and post-processing stages:
+
+- `ValueError`: Raised when a column contains values outside the allowed range for integers, when there are too many unique strings, or when encountering an unsupported data type.
+- `TypeError`: Raised when an operation is attempted on a data type that is not supported by the operation.
+
+## Example Workflow
+
+An example workflow where two clients encrypt two `DataFrame` objects, perform a merge operation on the server side, and then decrypt the results is available in the notebook [encrypted_pandas.ipynb](concrete-ml/use_case_examples/dataframe/encrypted_pandas.ipynb).
+
+## Current Limitations
+
+While this API offers a new secure way to work on remotely stored and encrypted data, it has some strong limitations at the moment:
+
+- **Precision of Values**: The precision for numerical values is limited to 4 bits.
+- **Supported Operations**: The `merge` operation is the only one available.
```

Comment:
more specifically : only left/right join , also only `how` and `on` parameters are supported